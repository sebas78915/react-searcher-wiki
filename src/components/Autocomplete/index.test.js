import React from 'react';
import { shallow } from 'enzyme';
import Autocomplete from '.';

const render = (props) => shallow(<Autocomplete {...props} />)

describe('Autocomplete component', () => {
  let sut;
  let props;

  describe('without props', () => {
    beforeEach(() => {
      sut = render(props);
    });

    it('Should match snapshot', () => {
      expect(sut).toMatchSnapshot();
    })
  })
  
  describe('When articles are passed', () => {
    beforeEach(() => {
      props = {
        articles: [1],
        searchValue: 'search value',
        onSearchChange: jest.fn()
      }
      sut = render(props);
    });

    it('Should match renderMenu snapshot', () => {
      const props = sut.props();

      expect(props.renderMenu()).toMatchSnapshot();
    })
  })

  describe('When articles are NOT passed', () => {
    beforeEach(() => {
      props = {
        articles: null,
        searchValue: 'search value',
        onSearchChange: jest.fn()
      }
      sut = render(props);
    });

    it('Should match renderMenu snapshot', () => {
      const props = sut.props();

      expect(props.renderMenu()).toMatchSnapshot();
    })
  })

  describe('When renderItem prop is called', () => {
    beforeEach(() => {
      sut = render(props);
    });

    describe('When item is highlighted', () => {
      it('Should match renderMenu snapshot', () => {
        const { renderItem } = sut.props();
        const item = {
          id: 1,
          label: 'test'
        }
  
        expect(renderItem(item, true)).toMatchSnapshot();
      })
    })

    describe('When item is NOT highlighted', () => {
      it('Should match renderMenu snapshot', () => {
        const { renderItem } = sut.props();
        const item = {
          id: 1,
          label: 'test'
        }
        expect(renderItem(item, false)).toMatchSnapshot();
      })
    })
    
    describe('When getItemValue is called', () => {
      beforeEach(() => {
        sut = render(props);
      });
      
      it('Should return label', () => {
        const { getItemValue } = sut.props();

        expect(getItemValue({label: 'test'})).toMatchSnapshot();
      })
    })
  })
});