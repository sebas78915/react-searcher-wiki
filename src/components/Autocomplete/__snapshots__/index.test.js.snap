// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete component When articles are NOT passed Should match renderMenu snapshot 1`] = `<React.Fragment />`;

exports[`Autocomplete component When articles are passed Should match renderMenu snapshot 1`] = `
<div
  className="input-suggestions"
  style={Object {}}
>
  <a
    className="search-link"
    href="/search?query=undefined"
  >
    See all results
  </a>
</div>
`;

exports[`Autocomplete component When renderItem prop is called When getItemValue is called Should return label 1`] = `"test"`;

exports[`Autocomplete component When renderItem prop is called When item is NOT highlighted Should match renderMenu snapshot 1`] = `
<div
  className="input-suggestions-item"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <a
    href={1}
    rel="noreferrer"
    target="_blank"
  >
    test
  </a>
</div>
`;

exports[`Autocomplete component When renderItem prop is called When item is highlighted Should match renderMenu snapshot 1`] = `
<div
  className="input-suggestions-item"
  style={
    Object {
      "backgroundColor": "#eee",
    }
  }
>
  <a
    href={1}
    rel="noreferrer"
    target="_blank"
  >
    test
  </a>
</div>
`;

exports[`Autocomplete component without props Should match snapshot 1`] = `
<Autocomplete
  autoHighlight={true}
  getItemValue={[Function]}
  inputProps={
    Object {
      "placeholder": "Input a search term...",
    }
  }
  isItemSelectable={[Function]}
  menuStyle={
    Object {
      "background": "rgba(255, 255, 255, 0.9)",
      "borderRadius": "3px",
      "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
      "fontSize": "90%",
      "maxHeight": "50%",
      "overflow": "auto",
      "padding": "2px 0",
      "position": "fixed",
    }
  }
  onChange={[Function]}
  onMenuVisibilityChange={[Function]}
  onSelect={[Function]}
  renderInput={[Function]}
  renderItem={[Function]}
  renderMenu={[Function]}
  selectOnBlur={false}
  value=""
  wrapperProps={Object {}}
  wrapperStyle={
    Object {
      "display": "inline-block",
    }
  }
/>
`;
